### PR TITLE
Prevent the mentor from applying for himself

### LINKED ISSUE #65: Prevent user from applying as both mentee and mentor for the same program
**Is your feature request related to a problem? Please describe.**
- Currently, A user can apply as both mentee and a mentor for the same program.

**Describe the solution you'd like**
- Remove the apply button of the programs tab if I am a mentor of that program

### PR DESCRIPTION
**Pull request by anjula-sack**
## Purpose

- The purpose of this PR is to fix #65 

## Goals

- Prevent the mentor from applying for himself

## Approach

- Hide the apply as a mentee button for the mentor

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by Gravewalker666** [comment]
@anjula-sack I think a mentor of a certain program shouldn't be allowed to apply as a mentee for the same program at all, just preventing the mentor from applying to himself implies that he can be a mentee of another mentor, which is absurd imo.

File: src/scenes/Home/components/ActivePrograms/index.tsx [hunk 1]
```diff
@@ -67,35 +91,38 @@ function ActivePrograms() {
...
-                        </Button>
-                      ) : (
-                        ''
                       )}
+                      {program.state == 'MENTEE_APPLICATION' &&
+                        !isUserAdmin &&
+                        mentoringPrograms &&
+                        !mentoringPrograms.find(
+                          (mentoringProgram: SavedProgram) =>
+                            mentoringProgram.id === program.id
```
Comment (Gravewalker666) [inline src/scenes/Home/components/ActivePrograms/index.tsx:107]: Is this check for `mentoringPrograms` necessary? 🤔 
What would happen if the `mentoringPrograms` array is empty?

Comment (anjula-sack) [reply src/scenes/Home/components/ActivePrograms/index.tsx:107]: Checking for `mentoringPrograms.length` is better I guess? If the arrays is empty the condition will be `false`.

Comment (Gravewalker666) [reply src/scenes/Home/components/ActivePrograms/index.tsx:107]: Let's say that there is a person who needs to apply as a mentee but not mentoring any programs, then his mentoringPrograms array would be empty right?

Comment (Gravewalker666) [reply src/scenes/Home/components/ActivePrograms/index.tsx:107]: With this logic, it implies that to be able to apply as a mentee in a certain program you have to be a mentor of at least one program right?

Comment (Gravewalker666) [reply src/scenes/Home/components/ActivePrograms/index.tsx:107]: I looked into it a bit, it was my bad sorry :) 
This is just a null check to see if the variable is initialized or not right?
I'll merge the PR.

**Review by Gravewalker666 — APPROVED** [review]

**Pull request merged**